src/server/proxies/MpsProxy.ts

```diff
@@ -0,0 +1,397 @@
+              //  target port
+              common.IntToStr(ciraChannel.channelid) +
+
+              common.IntToStr(data.length) +
+
+              data, 'binary')
+          )
+        } catch (error) {
+          log.error(`SetupCiraChannel write exception: ${error}`)
+        }
+      },
+      sendBuffer: undefined,
+      amtchannelid: undefined,
+      close: () => {
+        log.silly(`channel close for ${ciraChannel.nodeId}`)
+        this.WriteCloseChannel(ciraChannel)
+      },
+      closing: undefined,
+      onStateChange: undefined,
+      onData: undefined,
+      nodeId: uuid
+    }
+
+    // Send CHANNEL_OPEN msg to webProxy.
+    this.WriteSetupChannel(ciraChannel, targetport)
+
+    this.channels[currChannel] = ciraChannel
+    return this.channels[currChannel]
+  }
+
+  WriteSetupChannel (ciraChannel, targetport): void {
+    try {
+      log.silly('entering WriteSetupChannel')
+      this.socket.write(
+        Buffer.from(
+          // message header length
+          common.IntToStr(MPS_MESSAGE_HEADER_LENGTH) +
+          // Channel Open
+          String.fromCharCode(APFProtocol.CHANNEL_OPEN) +
+          // UUID
+          common.rstr2hex(ciraChannel.nodeId) +
+          //  target port
+          common.IntToStr(ciraChannel.channelid) +
+          //  Channelid
+          common.IntToStr(targetport), 'binary')
+      )
+    } catch (error) {
+      log.error(`WriteSetupChannel exception: ${error}`)
+    }
+  }
+
+  WriteCloseChannel (ciraChannel): void {
+    try {
+      log.silly('entering WriteCloseChannel')
+      this.socket.write(
+        Buffer.from(
+          // message header length
+          common.IntToStr(MPS_MESSAGE_HEADER_LENGTH) +
+          // Channel Open
+          String.fromCharCode(APFProtocol.CHANNEL_CLOSE) +
+          // UUID
+          common.rstr2hex(ciraChannel.nodeId) +
+          //  target port
+          common.IntToStr(ciraChannel.channelid), 'binary')
+      )
+    } catch (error) {
+      log.error('WriteCloseChannel exception')
+    }
+  }
+
+  static async getProxySocket (mpsServerIP: string, mpsservice: MPSMicroservice): Promise<any> {
+    // Global is a list of Global static variables
+    if (typeof MpsProxy.sockets[mpsServerIP] === 'undefined' || MpsProxy.sockets[mpsServerIP].destroyed) {
+      log.silly(`Create socket for ${mpsServerIP}`)
+      const mpsServerSocket: any = new net.Socket()
+      mpsServerSocket.setEncoding('binary')
+      // connect on the device
+
+      mpsServerSocket.connect(mpsservice.config.web_proxy_port, mpsServerIP, function () {
+        log.silly(`Connected to mpsServer on socket at IP ${mpsServerIP}`)
+      })
+
+      mpsServerSocket.on('error', (err) => {
+        log.error(`mps proxy error: ${err}`)
+        if (!mpsServerSocket.destroyed) {
+          mpsServerSocket.destroy()
+          log.silly('MPS Server Socket destroyed.')
+        }
+        // delete MpsProxy.proxies[guid]
+        delete MpsProxy.sockets[mpsServerIP]
```

Comment:
not for this PR -- but eventually we'll want to refactor out any delete statements due to performance hits.  This will likely become an issue when we scale with lots of connections